Fixed #35767 -- Adjusted customizing user model docs. (#18566)
django/django: docs/topics/auth/customizing.txt

Using a custom user model when starting a project
-------------------------------------------------

> If you're starting a new project, you can set up a custom user model, that
> behaves identically to the default user model by subclassing
> :class:`~django.contrib.auth.models.AbstractUser`::

Reviewer: I like this change. But, considering the PR title, I feel entitled to suggest that we consider going one step further and take the initiative to not have the annoying `pass` class definition in the example. This means being more tutorial-ly and less reference-y (and I think it's fine since we are in "topics").
```suggestion
If you're starting a new project, you can set up a custom user model that
behaves identically to the default user model by subclassing
:class:`~django.contrib.auth.models.AbstractUser`. For example:

.. code-block:: python
    :caption: ``myapp/models.py``

    from django.contrib.auth.models import AbstractUser
    from django.db import models
    from django.db.models import Case, Value, When
    from django.db.models.functions import Concat

    class User(AbstractUser):
        display_name = models.GeneratedField(
            expression=Case(
                When(first_name="", then="last_name"),
                When(last_name="", then="first_name"),
                default=Concat("first_name", Value(" "), "last_name"),
            ),
            output_field=models.TextField(),
            db_persist=False,
        )
```

Author: This no longer _"behaves identically to the default user model"_. 

I wouldn't add this, personally. `display_name` here is just duplicating `get_full_name()`, and in so doing leveraging the very `first_name`, `last_name` fields that we want to get rid of.

Author: TBH I don't want to add anything here. 

I think custom user models should just fade away slowly as we work on auth.User. 

I think by showing more advanced examples we just encourage people into what I consider a mistake. 

I also think adding content here adjusts to scope of what I've said we should implement here, which has wide agreement as far as I can see, and into territory where opinions will begin to differ, and which I've not argued for, and for which I'm not sure we've agreed a route forward. 

I don't want to flex the scope like that. I'd rather keep it as small as possible. 

Does that make sense?

Author: Yes. I don't think the existing docs are that great. 

I just don't want to rewrite them at length here. 🫠

Reviewer: Fair enough! Merging then.

Reviewer: (Well I will first push the removal of the comma before "that" because I feel that reads weird as proposed...)

Author: Thanks @nessita 🙏

-------------------------------------------

Changing :setting:`AUTH_USER_MODEL` after you've created database tables is

> possible, but somewhat involved, since it affects foreign keys and many-to-many

Reviewer 2: Trying to think about this from the perspective of a person using English as a foreign language & pondering the use of "involved". Maybe "complex" might convey the non-trivial nature of swapping

> possible, but somewhat involved, since it affects foreign keys and many-to-many

Author: In that case I’d go for maybe:

```suggestion
possible, but can be complex, since it affects foreign keys and many-to-many
```
